checks/prometheus_check.py

```diff
@@ -284,12 +338,45 @@ def process_metric(self, message, send_histograms_buckets=True, custom_tags=None
         `send_histograms_buckets` is used to specify if yes or no you want to send the buckets as tagged values when dealing with histograms.
         """
         try:
+            # If targeted metric, store labels
+            if message.name in self.label_joins:
+                matching_label = self.label_joins[message.name]['label_to_match']
+                for metric in message.metric:
+                    labels_list = []
+                    matching_value = None
+                    for label in metric.label:
+                        if label.name == matching_label:
+                            matching_value = label.value
+                        elif label.name in self.label_joins[message.name]['labels_to_get']:
+                            labels_list.append((label.name, label.value))
+                    if matching_label in self._label_mapping:
+                        self._label_mapping[matching_label][matching_value] = labels_list
+                    elif matching_value is not None:
+                        self._label_mapping[matching_label] = {matching_value : labels_list}
+
             if message.name in self.ignore_metrics:
                 return  # Ignore the metric
-            if message.name in self.metrics_mapper:
-                self._submit(self.metrics_mapper[message.name], message, send_histograms_buckets, custom_tags)
-            else:
-                getattr(self, message.name)(message, **kwargs)
+
+            # Filter metric to see if we can enrich with joined labels
+            if len(self.label_joins) > 0:
+                for metric in message.metric:
+                    for label in metric.label:
+                        if label.name in self._watched_labels:
+                            # Set this label value as active
+                            if label.name not in self._active_label_mapping:
+                                self._active_label_mapping[label.name] = {}
+                            self._active_label_mapping[label.name][label.value] = True
+                            # If mapping found add corresponding labels
+                            if label.name in self._label_mapping and label.value in self._label_mapping[label.name]:
+                                for label_tuple in self._label_mapping[label.name][label.value]:
+                                    extra_label = metric.label.add()
+                                    extra_label.name, extra_label.value = label_tuple[0], label_tuple[1]
```

Comment:
This block could be simplified by dropping the condition:

```python
                            try:
                                for label_tuple in self._label_mapping[label.name][label.value]:
                                    extra_label = metric.label.add()
                                    extra_label.name, extra_label.value = label_tuple[0], label_tuple[1]
                            except KeyError:
                                pass
```
What do you think ? 

> it's easier to ask for forgiveness than permission